tools/installer/full-node.sh

```diff
@@ -138,37 +195,62 @@ setup_poktrolld() {
-    # We need to start syncing from scratch using the version that was used when the network started.
-    POKTROLLD_VERSION=$(curl -s https://raw.githubusercontent.com/pokt-network/pocket-network-genesis/master/shannon/alpha/testnet-validated.init-version)
+    # Extract the version from genesis.json using jq
+    POKTROLLD_VERSION=$(jq -r '.app_version' < "$GENESIS_FILE")
+    print_color $YELLOW "Detected version from genesis: $POKTROLLD_VERSION"
+
+    if [ -z "$POKTROLLD_VERSION" ]; then
+        print_color $RED "Failed to extract version information from genesis file."
+        exit 1
+    fi
 
-    # Use the direct download link for the correct release
-    RELEASE_URL="https://github.com/pokt-network/poktroll/releases/download/${POKTROLLD_VERSION}/poktroll_linux_${ARCH}.tar.gz"
+    # Construct the release URL with proper version format
+    RELEASE_URL="https://github.com/pokt-network/poktroll/releases/download/v${POKTROLLD_VERSION}/poktroll_linux_${ARCH}.tar.gz"
+    print_color $YELLOW "Attempting to download from: $RELEASE_URL"
 
+    # Download and extract directly as the POKTROLL_USER
     sudo -u "$POKTROLL_USER" bash << EOF
     mkdir -p \$HOME/.poktroll/cosmovisor/genesis/bin
     curl -L "$RELEASE_URL" | tar -zxvf - -C \$HOME/.poktroll/cosmovisor/genesis/bin
+    if [ \$? -ne 0 ]; then
+        echo "Failed to download or extract binary"
+        exit 1
+    fi
     chmod +x \$HOME/.poktroll/cosmovisor/genesis/bin/poktrolld
     ln -sf \$HOME/.poktroll/cosmovisor/genesis/bin/poktrolld \$HOME/bin/poktrolld
     source \$HOME/.profile
 EOF
+
+    if [ $? -ne 0 ]; then
+        print_color $RED "Failed to set up Poktrolld"
+        exit 1
+    fi
+
     print_color $GREEN "Poktrolld set up successfully."
 }
 
 # Function to configure Poktrolld
 configure_poktrolld() {
     print_color $YELLOW "Configuring Poktrolld..."
     
-    # Ask for confirmation to download the genesis file
-    GENESIS_URL="https://raw.githubusercontent.com/pokt-network/pocket-network-genesis/master/shannon/alpha/testnet-validated.json"
-    print_color $YELLOW "The script will download the genesis file from:"
+    # Ask for confirmation to use the downloaded genesis file
+    print_color $YELLOW "The script has downloaded the genesis file from:"
     print_color $YELLOW "$GENESIS_URL"
-    read -p "Are you OK with downloading and using this genesis file? (y/N): " confirm_genesis
+    read -p "Are you OK with using this genesis file? (y/N): " confirm_genesis
     if [[ ! $confirm_genesis =~ ^[Yy] ]]; then
-        print_color $RED "Genesis file download cancelled. Exiting."
+        print_color $RED "Genesis file usage cancelled. Exiting."
         exit 1
     fi
 
+    # Detect external IP address
+    EXTERNAL_IP=$(curl -s https://api.ipify.org)
+    print_color $YELLOW "Detected external IP address: $EXTERNAL_IP"
+    read -p "Is this your correct external IP address? (Y/n): " confirm_ip
```

Comment:
Is anyone going to say no? Considering just commenting this out altogether

Reply:
I suspect there might be cases when the discovered IP address is not correct and requires adjustment.